src/site/topic8-nested.rst

```diff
@@ -0,0 +1,135 @@
+************************************
+Topic #8 Aside --- Nested Node Class
+************************************
+
+* Consider the doubly linked structure
+
+.. image:: img/links_double.png
+   :width: 500 px
+   :align: center
+
+* Since we need a reference to both the ``next`` and ``previous`` nodes, we can't use our existing ``Node`` :class:
+    * Simple to make, but still needs a change
+
+* Also consider that the ``Node`` class would only be used for a linked implementation of something
+    * As far as the user of a ``LinkedStack`` is concerned, they don't care about the ``Node`` class, they just care that the ``LinkedStack`` works
+    * Kinda like the private method ``expandCapacity`` in the ``ArrayStack``
+
+* Does it make sense to have the ``Node`` class accessible from everywhere?
+
+
+Nested Classes
+==============
+
+* We can put the ``Node`` class inside the ``LinkedStack`` class
+* If we do this, we can still access the ``Node`` class from the ``LinkedStack`` class
+* But, it's only accessible from within that class, so it keeps our ``Node`` class out of the way
+
+* Thinking about the doubly linked structure, what should I call the node for the, ``DoubleNode``?
+
```

Comment:
Packages also solve this, but a `Node` never leaves the boundaries of the "owning" class. The two classes (linked whatever and `Node`) are inextricably coupled, so putting them in the same place just kinda makes sense.

Reply:
_inextricably_